dds/DCPS/TypeAssignability.h

```diff
@@ -0,0 +1,137 @@
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#ifndef OPENDDS_DCPS_TYPE_ASSIGNABILITY_H
+#define OPENDDS_DCPS_TYPE_ASSIGNABILITY_H
+
+#include "dds/DCPS/TypeObject.h"
+
+#include <utility>
+#include <map>
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace XTypes {
+
+  class TypeLookup {
+  public:
+    const MinimalTypeObject& lookup_minimal(const TypeIdentifier& ti) const
+    {
+      return tmp_;
+      //      return table_[ti];
+    }
+
+  private:
+    MinimalTypeObject tmp_;
+    //    std::map<TypeIdentifier, MinimalTypeObject> table_;
+  };
+
+  // Set of pairs of members with each pair contains members from
+  // two structure types that have the same member ID and name
+  typedef std::vector<std::pair<const MinimalStructMember*,
```

Comment:
Use `OPENDDS_VECTOR` so we get allocator support in safety profile builds.

Reply:
I've updated this. In general, should I always use the macros defined for OpenDDS when working with containers?

Reply:
Yes, it's needed when the code in question is part of the core libraries that will be built when Safety Profile is enabled.  It's not a problem to also use in other contexts.